packages/lib/modules/transactions/transaction-steps/useTransactionSteps.tsx

```diff
@@ -40,6 +40,9 @@ export function useTransactionSteps(steps: TransactionStep[] = [], isLoading = f
   function resetTransactionSteps() {
     setCurrentStepIndex(0)
     setOnSuccessCalled({})
+    steps.forEach(step => {
+      if (step.transaction) resetTransaction(step.transaction)
+    })
```

Comment:
w/o this the steps seem to retain transaction data which we are relying on to save pool address to LS

If user were to:
1. create pool 
2. reset progress
3. start creating another pool

the pool address is added to local storage early because the `transaction?.execution?.data` exists which triggers the first "create pool" step being skipped

https://github.com/balancer/frontend-monorepo/blob/dedb7ca0a57dd83d0a7dc956f77b9d2ba2a782b6/packages/lib/modules/lbp/steps/useCreateLbpStep.tsx#L56-L65